ui/app/controllers/jobs/index.js

```diff
@@ -28,11 +51,95 @@ export default Controller.extend(Sortable, Searchable, {
   fuzzySearchProps: computed(() => ['name']),
   fuzzySearchEnabled: true,
 
+  optionsType: computed(() => [
+    { key: 'batch', label: 'Batch' },
+    { key: 'parameterized', label: 'Parameterized' },
+    { key: 'periodic', label: 'Periodic' },
+    { key: 'service', label: 'Service' },
+    { key: 'system', label: 'System' },
+  ]),
+
+  optionsStatus: computed(() => [
+    { key: 'pending', label: 'Pending' },
+    { key: 'running', label: 'Running' },
+    { key: 'dead', label: 'Dead' },
+  ]),
+
+  optionsDatacenter: computed('visibleJobs.[]', function() {
+    const flatten = (acc, val) => acc.concat(val);
+    const allDatacenters = new Set(
+      this.get('visibleJobs')
+        .mapBy('datacenters')
+        .reduce(flatten, [])
+    );
+
+    // Remove any invalid datacenters from the query param/selection
+    const availableDatacenters = Array.from(allDatacenters).compact();
+    scheduleOnce('actions', () => {
+      this.set(
+        'qpDatacenter',
+        qpSerialize(intersection(availableDatacenters, this.get('selectionDatacenter')))
```

Comment:
That'd work fine, but it wouldn't save me a dependency since I'm already using `intersection` elsewhere.

Reply:
Ah cool, I think it's just here:

https://github.com/hashicorp/nomad/blob/c12bdcdf0bde2d07d13dbd6e9865a3b46b8dba02/ui/app/models/allocation.js#L73

You could remove lodash there also? I suppose it depends what side of the fence you are on, but it would save a dependency, no biggie either way.